pkg/kubectl/cmd/wait/wait.go

```diff
@@ -315,24 +322,25 @@ func IsDeleted(info *resource.Info, o *WaitOptions) (runtime.Object, bool, error
 			}
 			return gottenObj, false, errWaitTimeoutWithName
 		default:
+			watchErr, ok := err.(*watchtools.WatcherError)
+			if ok {
+				fmt.Fprintf(w.ErrOut, "error: An error occurred while waiting for the condition to be satisfied: %v", watchErr)
+				continue
+			}
+
 			return gottenObj, false, err
 		}
 	}
 }
 
-// Wait has helper methods for handling watches, including error handling.
-type Wait struct {
-	errOut io.Writer
-}
-
 // IsDeleted returns true if the object is deleted. It prints any errors it encounters.
-func (w Wait) IsDeleted(event watch.Event) (bool, error) {
+func (w Wait) IsDeletedCond(event watch.Event) (bool, error) {
 	switch event.Type {
 	case watch.Error:
```

Comment:
It seems the UntilWithRetry will return the golang error before calling the conditionFn in case the event.type==watch.Error. Is this case still useful?